### PR TITLE
Validate single resource

### LINKED ISSUE #221: Validate single resource with `datapackage.json`
Is it possible to validate an individual resource file (CSV) against the schema in `datapackage.json`.

I have separate processing scripts for each resource and like to run `goodtables` for each individually.

Setting `schema` didn't work (as `datapackage.json` is not a Table schema I assume).

### PR DESCRIPTION
**Pull request by shashigharti**
* Added code to handle single resource validation
* Added tests

- fixes #221

### CODE REVIEW

File: frictionless/actions/validate.py [hunk 1]
```diff
@@ -103,6 +105,12 @@ def validate_package(
     # Create state
     timer = helpers.Timer()
 
+    # Read resource name for single resource validation
+    resource_name = None
```
Comment (roll) [inline frictionless/actions/validate.py]: I was wondering if we can just use a simpler logic:
- if `resource_name` is provided -> return from `resource.validate` function for this resource

Not sure if it's possible regarding options etc

File: tests/actions/validate/test_package.py [hunk 2]
```diff
@@ -532,3 +532,8 @@ def test_validate_package_with_resource_data_is_a_string_issue_977():
     assert report.flatten() == [
         [None, None, None, "package-error"],
     ]
+
+
+def test_validate_package_single_resource_221():
```
Comment (roll) [inline tests/actions/validate/test_package.py:536]: Let's add more tests (invalid resource/etc). Maybe edge case like wrong resource name

**Review by roll — CHANGES_REQUESTED** [review]
Looks good in-general!

Sorry I didn't write it but we also need to copy the implementation/test to `package.validate` function. Currently, there is a duplication till we release v5

File: frictionless/actions/validate.py (hunk 1 above)
Comment (shashigharti) [reply frictionless/actions/validate.py]: I have tried revising the code logic:

- The field_name has to be removed from options else it generates error in [line 123](https://github.com/frictionlessdata/frictionless-py/pull/1112/commits/921203d6c461e1991a15f848d17dce74a9ed4132#diff-08a6fec69d71afaa32a62bdda421723363f8235f66a588d3676dc717d639bb9dR123):


 However I could not think of any other solution for the filtering part of resources because the resources is being used in many different places:
- Validating
- Task error generation (parallel and sequential)

So I put it in a different helper function .

File: tests/actions/validate/test_package.py (hunk 2 above)
Comment (shashigharti) [reply tests/actions/validate/test_package.py:536]: I added tests for invalid resource name only. 
`[['task-error', 'The task has an error: No resource found with name ']]`

I could not think of other test case scenarios. We have already handled these:

- CLI  no resource name provided
- Valid resource provided
- Invalid resource name provided

Need some suggestions on it :)

**Comment by shashigharti** [comment]
Thank you! @roll I have made the changes. It is ready for the review.

**Review by roll — APPROVED** [review]
Thanks!

**Comment by roll** [comment]
@shashigharti it seems to be failing

Also, can you please add the same tests to `tests/package/validate` tomorrow?

**Comment by shashigharti** [comment]
@roll thank you! I have added new tests and fixed failing tests. It is ready for the review!.

**Review by roll — APPROVED** [review]
Thanks!
